### ISSUE TITLE
SSE specialization of the residfp-builder convolve function leads to audible artefacts (ringing)

### ISSUE THREAD

**Issue by reFX-Mike** [libsidplayfp/libsidplayfp#105]
This code-block (SSE implementation of the SincResampler of residfp-builder) is broken. At first it seems to work just fine, but very often, after a non-looping tune has finished playing, there are multiple frequencies still ringing.

https://github.com/libsidplayfp/libsidplayfp/blob/6d24872c8aa602c2f3d1caee321428d46450979b/src/builders/residfp-builder/residfp/resample/SincResampler.cpp#L103-L141

After benchmarking I found that if I just removed that piece of code and leave only these lines and remove the entire rest of the convolved function, everything works just fine and performs better!

https://github.com/libsidplayfp/libsidplayfp/blob/6d24872c8aa602c2f3d1caee321428d46450979b/src/builders/residfp-builder/residfp/resample/SincResampler.cpp#L239-L247

**Comment by mywave82** [comment]
I am impressed that you located this issue. I did some local testing of SSE version vs non-SSE version and I believe the linked PR fixes the problem.

**Comment by reFX-Mike** [comment]
I can confirm that it works correctly now.

I've replaced the entire convolve function with this:

```
auto convolve = [] ( const int16_t* __restrict__ a, const int16_t* __restrict__ b, int bLength )
{
	auto    out = 0;

	while ( bLength-- )
		out += *a++ * *b++;

	return ( out + ( 1 << 14 ) ) >> 15;
};
```

It's only used twice within the same function and can't be called from the outside, so I decided to make it a local lambda and remove all specializations. The performance is identical across hundreds of minutes of audio rendered at 44.1 kHz/mono. The specializations are made obsolete by modern compilers.

For the sake of readability and maintainability, I would replace them with the short C++ loop.
